packages/unlock-js/src/erc20.ts

```diff
@@ -93,6 +93,96 @@ export async function approveTransfer(
   return contract.approve(lockContractAddress, value)
 }
 
+interface TransferAuthorizationMessage {
+  from: string
+  to: string
+  value: any
+  validAfter: number
+  validBefore: number
+  nonce: string
+}
+
+export async function signTransferAuthorization(
```

Comment:
I am not sure this is supported by many ERC20 but I was not sure what the best approach was to keep things simple.